File: Algorithms/038.CountAndSay.java
public class Solution {
    public String countAndSay(int n) {
        List<Integer> v = new ArrayList<>();
        List<Integer> w = new ArrayList<>();
        String result = "";
        v.add(1);
        while (--n > 0) {
            int count = 1;
            for (int i=1; i < v.size(); ++i)
            {
                if (v.get(i) == v.get(i-1)) {
                    count ++;
                }
                else {
                    w.add(count);
                    w.add(v.get(i-1));
                    count = 1;
                }
            }
            w.add(count);
            w.add(v.get(v.size()-1));
            v = w;
            w = new ArrayList<>();
        }
        for (int i = 0; i < v.size(); ++i)
        {
            result += v.get(i);
        }
        return result;
    }
}

File: Algorithms/cpp/038.CountAndSay.cpp
class Solution {
public:
    string countAndSay(int n) {
        vector<int> v;
        vector<int> w;
        string result;
        v.push_back(1);
        while (--n) {
            int count = 1;
            for (int i=1; i < v.size(); ++i)
            {
                if (v[i] == v[i-1]) {
                    count ++;
                }
                else {
                    w.push_back(count);
                    w.push_back(v[i-1]);
                    count = 1;
                }
            }
            w.push_back(count);
            w.push_back(v[v.size()-1]);
            v = w;
            w.clear();
        }
        for (int i = 0; i < v.size(); ++i)
        {
            result += v[i] + '0';
        }
        return result;
    }
};
